core/src/main/java/de/jplag/merging/MatchMerging.java

```diff
@@ -0,0 +1,136 @@
+package de.jplag.merging;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import de.jplag.JPlagComparison;
+import de.jplag.JPlagResult;
+import de.jplag.Match;
+import de.jplag.Submission;
+import de.jplag.Token;
+import de.jplag.options.JPlagOptions;
+
+/**
+ * This class implements match merging against obfuscation. Based on configurable parameters MergeBuffer and
+ * SeperatingThreshold, it alters prior results and merges all neighboring matches that fit the specified thresholds.
+ * When neighboring matches get merged they become one and the tokens separating them get removed from the submission
+ * clone. MergeBuffer describes how shorter a match can be than the Minimum Token Match. SeperatingThreshold describes
+ * how many tokens can be between two neighboring matches. Both are set in {@link JPlagOptions} as
+ * {@link MergingParameters} and default to 0 (which deactivates merging).
+ */
+public class MatchMerging {
+    private int minimumTokenMatch;
+    private Submission firstSubmission;
+    private Submission secondSubmission;
+    private List<Match> globalMatches;
+    private List<List<Match>> neighbors;
+    private int seperatingThreshold;
+    private JPlagResult result;
+    private List<JPlagComparison> comparisons;
+    private JPlagOptions options;
+
+    /**
+     * Constructor for class MatchMerging
+     * @param result is the initially computed result object
+     * @param options encapsulates the adjustable options
+     */
+    public MatchMerging(JPlagResult result, JPlagOptions options) {
+        this.result = result;
+        this.comparisons = new ArrayList<>(result.getAllComparisons());
+        this.options = options;
+        this.minimumTokenMatch = options.minimumTokenMatch();
+        this.seperatingThreshold = options.mergingParameters().seperatingThreshold();
```

Comment:
Both of these fields are only used once, but you also keep the options as fields. Remove both fields and access options directly.